### PR TITLE
BLD: only compile fixed_interpolator.cpp once

### LINKED ISSUE #4278: BLD: Fails to build: internal_error
### Bug report

**Bug summary**

When building the package as part of the packaging `yt` for openSUSE, the fails often (not always, something between 50 % and 80 %) it fails with an error like this:

```
[  338s]   gcc -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=r
eturn-type -g -DOPENSSL_LOAD_CONF -fwrapv -fno-semantic-interposition -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protecti
on -Werror=return-type -g -IVendor/ -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -IVendor
/ -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -freport-bug -fPIC -Iyt/utilities/
lib/ -Iyt/utilities/lib -I./yt/utilities/lib -I/usr/include/python3.10 -I/usr/lib64/python3.10/site-packages/numpy/core/include -c yt/utilities/lib/marching_cubes.cpp -o build/temp.linux-x86_64-cpython-310/yt/u
tilities/lib/marching_cubes.o
[  338s]   lto1: internal compiler error: resolution sub id 0xadc1906a211a636c not in object file
[  338s]   0x102f76b internal_error(char const*, ...)
[  338s]         ???:0
[  338s]   0x15cfb50 lto_main()
[  338s]         ???:0
[  338s]   Please submit a full bug report, with preprocessed source.
[  338s]   Please include the complete backtrace with any bug report.
[  338s]   See <https://bugs.opensuse.org/> for instructions.
[  338s]   lto-wrapper: fatal error: /usr/bin/g++ returned 1 exit status
[  338s]   compilation terminated.
[  338s]   /usr/lib64/gcc/x86_64-suse-linux/12/../../../../x86_64-suse-linux/bin/ld: error: lto-wrapper failed
[  338s]   collect2: error: ld returned 1 exit status
```

**Code for reproduction**

It gets triggered while building the package, so no obvious code snippet.

[Complete build log](https://github.com/yt-project/yt/files/10331308/_log.txt) with all packages used and steps taken to achieve this problem.

**Actual outcome**

See the build error above.

**Expected outcome**

Clean build.

**Version Information**
  * Operating System: Linux/openSUSE/Tumbleweed (rolling distro) as of 2023-01-02
  * Python Version: this one is for Python 3.10.9, but I don’t think it is interpreter driven
  * yt version: 4.1.2 from the PyPI tarball
  * Other Libraries (if applicable): see above log for the complete list

All packages from openSUSE packages.

----

When I have consulted this with our GCC expert (hi, @marxin!) I got this suggestion:

> Hi, I'm almost sure that the problem is that there are some wrong dependencies and some .o file is used before its producer completes it. I have seen it many times, it manifests itself in different voices and the fact that it is not deterministic:
> 
> [   10s]   lto1: internal compiler error: resolution sub id 0xb2bc78268f7df7a0 not in object file
> [   10s]   0x102f987 internal_error(char const*, ...)
> [   10s]         ???:0
> [   10s]   0x15d02f0 lto_main()
> [   10s]         ???:0
> 
> or that message of yours with lto1: internal compiler error: original not compressed with zstd

### PR DESCRIPTION
**Pull request by yut23**
## PR Summary

Avoid a race condition on fixed_interpolator.o during parallel builds by building it only once and storing it in a static library.

Hopefully fixes #4278.

## PR Checklist

- [ ] New features are documented, with docstrings and narrative docs
- [ ] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

File: setupext.py [hunk 1]
```diff
@@ -368,6 +369,31 @@ def _compile(
     CCompiler.compile = _compile
 
 
+def get_python_include_dirs():
+    """Extracted from distutils.command.build_ext.build_ext#finalize_options(),
```
Comment (neutrinoceros) [inline setupext.py:373]: ?
```suggestion
    """Extracted from distutils.command.build_ext.build_ext.finalize_options(),
```

**Review by neutrinoceros — APPROVED** [review]
This looks good as far as I can tell, thank you very much for digging into it !

File: setupext.py (hunk 1 above)
Comment (yut23) [reply setupext.py:373]: Oh, that's just a convention I picked up from Java (I think?) to distinguish between free functions and object methods.

Comment (neutrinoceros) [reply setupext.py:373]: I'm not sure it makes sense outside of Java. At least I never encountered this before and it just looks like a typo to my untrained eye.

Comment (yut23) [reply setupext.py:373]: I'll get rid of it then.

**Review by cphyc — APPROVED** [review]

**Pull request merged**